frame/helpers.py

```diff
@@ -0,0 +1,36 @@
+import os
+from typing import Iterable
+
+def combine(*arg):
+    result = [()]
+    for input_list in arg:
+        assert isinstance(input_list, list), 'combine input not all lists'
+        partial_result = result
+        result = []
+        for partial_tuple in partial_result:
+            for input_item in input_list:
+                result.append(partial_tuple + (input_item,))
+    return result
```

Comment:
Couldn't this be [`itertools.product`](https://docs.python.org/3/library/itertools.html#itertools.product)?